swan-lake/development-tutorials/run-in-the-cloud/deployment-guide/ballerina-k8s-deployment-guide.md

```diff
@@ -0,0 +1,357 @@
+
+For a more complex deployment, this demonstrates a Ballerina app that reads a greeting from a config map and responds to HTTP requests. It covers Kubernetes deployment with resource limits, autoscaling, config maps, and health probes using Code to Cloud. 
+
+1. Create a new Ballerina package
+
+```bash
+$ bal new hello_k8s
+```
+2. Replace the `main.bal` file with the following content
+```ballerina 
+import ballerina/http;
+
+listener http:Listener helloEP = new(9090);
+
+configurable string greeting = ?;
+
+service /helloWorld on helloEP {
+    resource function get sayHello() returns string {
+        return greeting + ", Kubernetes!";
+    }
+}
+```
+3. Create `probes.bal` with health and readiness probes
+
+```ballerina
+import ballerina/http;
+
+listener http:Listener probeEP = new(9091);
+
+service /probes on probeEP {
+    resource function get healthz() returns boolean {
+        return true;
+    }
+    resource function get readyz() returns boolean {
+        return true;
+    }
+}
+```
+4. Create a `Config.toml` file
+
+```toml
+greeting = "Hello"
+```
+
+5. Update the `Cloud.toml` to include auto-scaling, resource limits, config maps, and probe configuration
+
+```toml
+[container.image]
+repository="wso2inc"
+name="hello-k8s"
+tag="v0.1.0"
+
+[cloud.deployment]
+min_memory="100Mi"
+max_memory="256Mi"
+min_cpu="500m"
+max_cpu="500m"
+
+[cloud.deployment.autoscaling]
+min_replicas=2
+max_replicas=5
+cpu=60
+
+[[cloud.config.files]]
+file="./Config.toml"
+
+[cloud.deployment.probes.liveness]
+port=9091
+path="/probes/healthz"
+
+[cloud.deployment.probes.readiness]
+port=9091
+path="/probes/readyz"
+```
+6. Build the package: 
+```
+$ bal build --cloud="k8s"
+  Compiling source
+       wso2/hello_k8s:0.1.0
+
+  Generating executable
+
+  Generating artifacts...
+
+       @kubernetes:Service                      - complete 1/2
+       @kubernetes:Service                      - complete 2/2
+       @kubernetes:ConfigMap                    - complete 1/1
+       @kubernetes:Deployment                   - complete 1/1
+       @kubernetes:HPA                          - complete 1/1
+       @kubernetes:Docker                       - complete 2/2 
+
+       Execute the below command to deploy the Kubernetes artifacts: 
+       kubectl apply -f /home/wso2/user/hello_k8s/target/kubernetes/hello_k8s
+
+       Execute the below command to access service via NodePort: 
+       kubectl expose deployment hello-k8s-deployment --type=NodePort --name=hello-k8s-svc-local
+
+       target/bin/hello_k8s.jar
+   ```
+> **Tip:** You can update `Ballerina.toml` with build option to integrate option to `bal build`
+> ```
+> [build-options] 
+>  cloud = "k8s"
+> ```
+   
+- Execute the Kubernetes service, Push Docker image to Docker Hub: 
+```
+ $ docker push wso2inc/hello-k8s:v0.1.0
+```
+- Build the GraalVM executable in Kubernetes
+
+  For deploying Ballerina applications on Kubernetes with GraalVM, the GraalVM build offers significant benefits in 
```

Comment:
The first sentence needs to be redone IMO. The second part is more a benefit of using GraalVM.

Reply:
Re-arranged the sentence, please review